supabase/models/upsertRow.ts

```diff
@@ -0,0 +1,17 @@
+import newClient from '../utils/newClient';
+import { PartialItem } from '@/types/supabaseTypes';
+
+export default async function upsertRow(
+  table: string,
+  upsertValues: PartialItem
+) {
+  const supabase = newClient();
+  const { error } = await supabase.from(table).upsert(upsertValues);
+
+  if (error) {
+    console.log(error);
+    return false;
+  }
+
+  return true;
```

Comment:
Hey @camelPhonso so I did this with the code:
```
  try {
    const { data: editedItem, error } = await supabase
      .from(table)
      .upsert(upsertValues);

    if (error) {
      console.log(error);
      throw error;
    }
    
    return editedItem;
  } catch (error) {
    console.log(error);
    throw error;
  }
  ```
But now I have a issue where "editedItem" is always null even if I refresh the page and I see that the data was edited. That is why before I returned "true". I can't find why or how to get a response of some sort to send back to say that the operation was successful.
And I added another if statement to log a "error" coming back from the database because I assume "catch (error)" and "error" from database would be different.

Reply:
So I found the issue it was missing ".select();" at the end of this:
```
    const { data: editedItem, error } = await supabase
      .from(table)
      .upsert(upsertValues)
      .select();
 ```
 now I get the data back from the database so all good.